docs/TuxReact.md

```diff
@@ -0,0 +1,449 @@
+
+***
+
+#### <a id="TuxedoJS-OwneeClass"></a>5.2) TuxedoJS OwneeClass [#](#TuxedoJS-OwneeClass)
+`Tux` provides an `OwneeClass` that is able to take advantage of its parent's `OwnerProps` to access these shared methods. All `Tux` classes are built off of the `OwneeClass`, which means that all `Tux` classes have access to these tools (including `OwnerClass` components):
+
+```
+    var React = require('tux/React');
+    var Todo = require('./Todo');
+
+    var Todos = React.createOwneeClass({
+      propTypes: {
+        todos: React.PropTypes.array.isRequired
+      },
+
+      //Tux components can specify needed ownerProps just as they can with standard React props
+      nearestOwnerPropTypes: {
+        addTodo: React.PropTypes.func.isRequired
+      },
+
+      handleSubmit: function (e) {
+        e.preventDefault();
+        var todoText = this.refs.todoText.getDOMNode();
+
+        this.nearestOwnerProps.addTodo({
+          text: todoText.value
+        });
+
+        todoText.value = '';
+      },
+
+      render: function () {
+        var TodoArray = this.props.todos.map(function (todo) {
+          return <Todo todo={todo} />
+        });
+
+        return (
+          <div>
+            <form onSubmit={this.handleSubmit}>
+              <input ref="todoText" />
+            </form>
+            { TodoArray }
+          </div>
+        );
+      }
+    });
+```
+
+The `OwneeClass` can automatically access the properties and methods exposed to it by its nearest `OwnerClass` through the `nearestOwnerProps` key. You can read more about it [here](tuxedojs.org/docs/TuxReact/TuxOwneeClass).
+
+***
+
+#### <a id="TuxedoJS-MutableClass"></a>5.3) TuxedoJS MutableClass [#](#TuxedoJS-MutableClass)
+`Tux` provides a `MutableClass` designed to take advantage of the performance boosting tools in `React`. Like the other `Tux` classes it to is built off of the `OwneeClass`:
+
+```
+    var React = require('tux/React');
+
+    var Todo = React.createMutableClass({
+      propTypes: {
+        todo: React.PropTypes.object.isRequired
+      },
+
+      //the MutableClass defines a set of mutableTraits which allow it to determine when it actually needs to be updated in the virtual DOM
+      mutableTraits: {
+        props: 'text'
+      },
+
+      render: function () {
+
+        return (
+          <div>
+            <h1>{ this.props.todo.text }</h1>
+          </div>
+        );
+      }
+    });
+```
+
+The `MutableClass` will use its `mutableTraits` key to find the `props` and `state` that can change. In this case, it will find the `props.todo.text` property. When the component reaches the `shouldComponentUpdate` lifecycle event it will check the old copy of this propery versus the new copy to determine if it should update automatically. You can read more about it [here](tuxedojs.org/docs/TuxReact/TuxMutableClass).
```

Comment:
Recommend mentioning that the MutableClass will find the props and state that can change on the specified mutable traits, that way it doesn't seem like it automatically knows which things can change.